src/util/sighash.rs

```diff
@@ -988,7 +997,7 @@ mod tests {
         let prevouts = if sighash_type.split_anyonecanpay_flag().1 && tx_bytes[0] % 2 == 0 {
             // for anyonecanpay the `Prevouts::All` variant is good anyway, but sometimes we want to
             // test other codepaths
-            Prevouts::One(input_index, &prevouts[input_index])
+            Prevouts::One(input_index, prevouts[input_index].clone())
```

Comment:
This clone was done because the generic `T` needed to be the same type across if else arms

Reply:
Doesn't matter much in test anyway.

Reply:
Yeah, just highlighted because it might confuse review.

Reply:
The `taproot_signature_hash` function also gets the correct data for sighash single from prevouts::All. I think it is required to extract the corresponding txout from an array in this fashion. This was only done to test that both of these work.